### PR TITLE
Add ZStream constructors for Java Iterators and Streams

### PR DESCRIPTION
**Pull request by iravid**

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/Stream.scala [hunk 1]
```diff
@@ -225,6 +226,30 @@ object Stream extends Serializable {
   final def fromIteratorManaged[E, A](iterator: Managed[E, Iterator[A]]): Stream[E, A] =
     ZStream.fromIteratorManaged(iterator)
 
+  /**
+   * See [[ZStream.fromJavaIterator]]
+   */
+  final def fromJavaIterator[E, A](iterator: IO[E, ju.Iterator[A]]): Stream[E, A] =
```
Comment (regiskuckaertz) [inline streams/shared/src/main/scala/zio/stream/Stream.scala]: Shouldn't these go under the `jvm` branch?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/Stream.scala]: Hmm it compiles on JS but that might be misleading. I’ll check if Iterator and Stream are implemented there.

**Comment by vasilmkd** [comment]
This looks like a lot of repetitive code. Could we maybe use some of the methods defined here and delegate to the old implementations of the iterator code?
https://www.scala-lang.org/api/2.13.1/scala/collection/JavaConverters$.html

**Comment by iravid** [comment]
Yes, good point @vasilmkd!

File: streams/shared/src/main/scala/zio/stream/Stream.scala (hunk 1 above)
Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/Stream.scala]: So there's a ju.Iterator implementation but not Stream. Weird that it compiles. Anyway I moved the Stream things to platform-specific.

Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/Stream.scala]: What about this? Not very familiar with converters though. Maybe you could take a look.
https://www.scala-lang.org/api/2.13.1/scala/jdk/StreamConverters$.html

Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/Stream.scala]: Ah, maybe it's only available for 2.13. :/

**Comment by iravid** [comment]
Dropped the use of the JavaConverters to avoid 2.12/2.13 compatibility shenanigans.

@vasilmkd StreamConverters only provides an eager transformation from Java's Stream to a Scala Collection, so not what we need here.

Merging when CI is green.

**Comment by vasilmkd** [comment]
TIL, thanks.

**Pull request merged**